### DISCUSSION TITLE
[Proposal] Add the way to set http-headers on mediaelement

### DISCUSSION THREAD

**Discussion by JieNiGui06** [CommunityToolkit/Maui#1889] [New Feature Discussions]
### Feature name

http-headers property

### Link to discussion

no

### Progress tracker

- [ ] Android Implementation
- [ ] iOS Implementation
- [ ] MacCatalyst Implementation
- [ ] Windows Implementation
- [ ] Tizen Implementation
- [ ] Unit Tests
- [ ] Samples
- [ ] Documentation

### Summary

set the http headers

### Detailed Design

no

### Usage Syntax

```markdown
meidaele.source = new xxx(new uri("xxx"),new xxxx{
{"referrer","abc"},
{"user-angent","def"}
});
```

**Comment by JieNiGui06** [comment]
I think very important.

**Comment by kapsiR** [comment]
This is a must-have! Otherwise you can't use any authentication header, like a Bearer token?!

**Comment by it11111111** [comment]
This is crucial

**Comment by bijington** [comment]
@kapsiR and @it11111111 we openly accept PRs to this repository, if it is something that is crucial or a must-have would either of you be willing to contribute such a feature?

**Reply by kapsiR** [reply]
@bijington May you guide us where to start here please? Is there any architecture overview that we have to consider?

**Reply by bijington** [reply]
Yes of course, we typically want to make sure that the community buys into feature requests and hope for at least 10 votes on the request before we proceed, I know this item doesn't but I will mark it for discussion for the next standup and see what the team think as I suspect it will be of value.

In the meantime we have the contributing guide hear: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md

The next step would be to come up with some level of API design to show how this would work for developers rather than the bare submission in this request but we can work on that together.

**Comment by Kaaybi** [comment]
For information, I implemented this here https://github.com/CommunityToolkit/Maui/pull/3169
